File: Sources/OptolithDatabaseSchema/GeneratedTypes/specialAbility/_Tradition.swift
//
//  _Tradition.swift
//  OptolithDatabaseSchema
//

public struct SpecialRule: EntitySubtype {
    /// An optional label that is displayed and placed before the actual text.
    public let label: NonEmptyString?

    /// The text of a special rule.
    public let text: NonEmptyString

    public init(label: NonEmptyString? = nil, text: NonEmptyString) {
        self.label = label
        self.text = text
    }
}

File: src/types/specialAbility/_Tradition.ts
import { Object, Optional, Required, String, TypeAlias } from "tsondb/schema/dsl"

export const SpecialRule = TypeAlias(import.meta.url, {
  name: "SpecialRule",
  type: () =>
    Object({
      label: Optional({
        comment: "An optional label that is displayed and placed before the actual text.",
        type: String({ minLength: 1 }),
      }),
      text: Required({
        comment: "The text of a special rule.",
        type: String({ minLength: 1 }),
      }),
    }),
})
